src/main/scala/is/hail/expr/ir/PointedTypeInfo.scala

```diff
@@ -0,0 +1,32 @@
+package is.hail.expr.ir
+
+import is.hail.asm4s._
+
+sealed trait PointedTypeInfo[T] {
+  def ti: TypeInfo[T]
+  def point: Code[_]
+}
+object PointedTypeInfo {
+  def pointedTypeInfo[T](implicit tpti: PointedTypeInfo[T]) = tpti
+  implicit case object PointedIntTypeInfo extends PointedTypeInfo[Int] {
+    val ti = IntInfo
+    val point = const(-2)
+  }
+  implicit case object PointedLongTypeInfo extends PointedTypeInfo[Long] {
+    val ti = LongInfo
+    val point = const(-3L)
+  }
+  implicit case object PointedFloatTypeInfo extends PointedTypeInfo[Float] {
+    val ti = FloatInfo
+    val point = const(-4.0f)
+  }
+  implicit case object PointedDoubleTypeInfo extends PointedTypeInfo[Double] {
+    val ti = DoubleInfo
+    val point = const(-5.0)
```

Comment:
This seems bad since it isn't a single instruction.

Reply:
Ah yes, I put in non-zero values so they would jump out in the printed byte code, I can change them to zeros.